main.go

```diff
@@ -76,13 +76,23 @@ func (s *serialPort) Write(b []byte) (int, error) {
 
 var deferredFuncs []func()
 
+// RegisterAgent is a blocking call, the RPC itself has retry logic baked it
+// with jitter and backoff up to a total of 10 minutes.
+// If client creation or register agent (after retries) fail we then wait for
+// 5 minutes and try again.
 func registerAgent(ctx context.Context) {
-	if agentconfig.TaskNotificationEnabled() || agentconfig.GuestPoliciesEnabled() {
-		if client, err := agentendpoint.NewClient(ctx); err != nil {
-			logger.Errorf(err.Error())
-		} else if err := client.RegisterAgent(ctx); err != nil {
-			logger.Errorf(err.Error())
+	for {
+		if agentconfig.TaskNotificationEnabled() || agentconfig.GuestPoliciesEnabled() {
```

Comment:
Curious to know why this check is required. Is this to handle older agents?

I think a comment might help here since it is not obvious why these checks are required.

Reply:
Updated, the logic here to move this check to the caller so this is clearer